data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoServiceError.java

```diff
@@ -9,21 +9,46 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/*
+  This class represents an error that returned from the service
+ */
 public class KustoServiceError extends Exception {
-    private ArrayList<Exception> exceptions = null;
+    private ArrayList<Exception> exceptions;
 
-    public KustoServiceError(JSONArray exceptions) throws JSONException {
+    public KustoServiceError(JSONArray jsonExceptions, boolean isOneApi, String message) throws JSONException {
+        this(message);
         this.exceptions = new ArrayList<>();
-        for (int j = 0; j < exceptions.length(); j++) {
-            this.exceptions.add(new Exception(exceptions.getString(j)));
+        if (isOneApi){
```

Comment:
Nit: I would move this if to *inside* the for loop, to eliminate the duplicated for loop for readability.

Isn't this less performant? JIT compiler to the rescue! When it sees loop-invariant conditions, it optimizes them to outside of the loop. So we get to have the best of both worlds - code that's more readable, and more performant.